### ISSUE TITLE
About the creation of the D-NeRF dataset

### ISSUE THREAD

**Issue by JianxiangL** [azzarelli/datacollection#1]
I downloaded the D-NeRF dataset, but I found that the images in the training set are not what I expected to be taken by a camera with continuous shooting. Instead, it seems that the camera motion between two consecutive images is not smooth and continuous. Could you help me understand this? I would really appreciate it!

**Comment by azzarelli** [comment]
Sure.

As a preliminary, its worth noting that the D-NeRF datasets is lightweight/sparse - one of the main accomplishments with the original NeRF was it's ability to reconstruct scenes from 20-30 images. The same is true for D-NeRF, where we have between 50 and 200 frames depending on which scene we use.

Anyway, let's start with the simple case when the camera is static but the object is dynamic. In this case there are going to be a lot of occlusions, for example if our static camera is placed in front of the D-NeRF T-Rex, we won't see the motion of it's tail. This is also going to be true if we move the camera slowly around the subject, perhaps for the first 3 seconds of the video we only see the front half of the T-Rex.

So to keep the dataset short (between 50-200 frames) while also minimizing occlusion, the views at different time points are placed at various random positions. This improves the geometric awareness of the model and makes it faster to train. As the D-NeRF model takes days to train, the point about training speed was very necessary for their model to succeed.

Also, its worth adding that this allows for fair testing/evaluation. If my data is a sequence of frames from a camera moving around a scene, then randomly selecting a frame and using it for testing is going to introduce a positional bias in the test set as the test frame is going to appear right next to the training frames. Hence, D-NeRF avoids this bias by randomly distributing training and testing views.

To be honest, it's best to view the D-NeRF dataset as an ideal but unrealistic dataset. It good for validating that a model does what it said it will do, but its not a good way to evaluate how good the model is in practice.

Hope this helps! Will close the comment if this has answered your questions!
